compiler/src/main/java/com/squareup/anvil/compiler/codegen/BindingModuleGenerator.kt

```diff
@@ -473,4 +497,32 @@ private sealed class GeneratedMethod {
   class BindingMethod(override val spec: FunSpec) : GeneratedMethod()
 }
 
+private data class Binding(
+  val contributedClass: ClassDescriptor,
+  val annotation: AnnotationDescriptor,
+  val boundType: ClassDescriptor
+)
+
+private fun List<Binding>.findHighestPriorityBinding(): Binding {
+  if (size == 1) return this[0]
+
+  val bindings = groupBy { it.annotation.priority() }
+    .filter { it.value.isNotEmpty() }
```

Comment:
When would this be empty?

Reply:
It was more like a sanity check, but you're absolutely right it shouldn't happen. With the tests the sanity check shouldn't be necessary. I removed this line.